gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/JdbcExtractor.java

```diff
@@ -1125,6 +1129,35 @@ private Schema getCustomColumnSchema(String columnName) {
   }
 
   /**
+   * Check if the SELECT query covers multiple tables
+   *
+   * <p>
+   *   The idea is to check if "," shows up in the "FROM" clause. The space before the next SQL keyword determines
+   *   the end of the "FROM" clause
+   * </p>
+   */
+  public static boolean hasMultipleTables(String selectQuery) {
+    if (selectQuery == null || selectQuery.length() == 0) {
+      return false;
+    }
+
+    String queryLowercase = selectQuery.toLowerCase();
+    int tableListStartIndex = queryLowercase.indexOf(" from ") + 6;
+
+    // Split the string after FROM clause by space(consecutive spaces as a single space)
+    final String spaceRegex = " +";
+    String[] items = queryLowercase.substring(tableListStartIndex).trim().split(spaceRegex);
+
+    // "From tableA,tableB"
+    if (items.length == 1) {
+      return items[0].contains(",");
+    }
+
+    // Cases like "FROM tableA, tableB <SQL KEYWORD>"
+    return items[0].contains(",") || items[1].contains(",");
```

Comment:
I think this doesn't handle the following case: select a.fromLoc from a aliasA , b aliasB